nodes/stampzilla-server/handlers/securewebsockethandler.go

```diff
@@ -179,6 +180,36 @@ func (wsh *secureWebsocketHandler) Message(s interfaces.MelodySession, msg *mode
 			return err
 		}
 		wsh.WebsocketSender.SendToID(node.UUID, "setup", node)
+	case "setup-device":
+		device := &devices.Device{}
+		err := json.Unmarshal(msg.Body, device)
+		if err != nil {
+			return err
+		}
+
+		logrus.WithFields(logrus.Fields{
+			"from":   msg.FromUUID,
+			"config": device,
+		}).Debug("Received new device configuration")
+
+		node := wsh.Store.GetNode(device.ID.Node)
+		if node == nil {
+			return fmt.Errorf("Node was not found")
+		}
+
+		if node.Aliases == nil {
+			node.Aliases = make(map[devices.ID]string)
+		}
+		node.Aliases[device.ID] = device.Alias
+
+		wsh.Store.AddOrUpdateNode(node)
+		err = wsh.Store.SaveNodes()
+		if err != nil {
+			return err
+		}
+
+		dev := wsh.Store.GetDevices().Get(device.ID)
+		wsh.Store.AddOrUpdateDevice(dev)
```

Comment:
`wsh.Store.AddOrUpdateDevice` is a bit overkill? Why not just dev.Alias = device.Alias? I guess we need some way to run devices callback so this might be the best solution for now. AddOrUpdateDevice could in the future be more smart and only apply the changes detected. Now you risk overwriting the entire device with perhaps wrong data if gui is not 100% up to date with node state. 

Would be nice if only alias was the thing updated on the device.